File: src/DatenMeister.SourceGeneration.Console/mof.cs
using DatenMeister.Core.EMOF.Interface.Reflection;
using DatenMeister.Provider.InMemory;

// Created by DatenMeister.SourcecodeGenerator.ClassTreeGenerator Version 1.1.0.0 created at 21.01.2017 22:32:50
namespace DatenMeister.Core
{
    public class _MOF
    {
        public class _Identifiers
        {
            public class _URIExtent
            {
            }

            public _URIExtent @URIExtent = new _URIExtent();
            public IElement @__URIExtent = new InMemoryElement();

            public class _Extent
            {
            }

            public _Extent @Extent = new _Extent();
            public IElement @__Extent = new InMemoryElement();

        }

        public _Identifiers Identifiers = new _Identifiers();

        public class _EMOF
        {
        }

        public _EMOF EMOF = new _EMOF();

        public class _CMOFExtension
        {
            public class _Tag
            {
                public static string @tagOwner = "tagOwner";
                public IElement _tagOwner = null;

            }

            public _Tag @Tag = new _Tag();
            public IElement @__Tag = new InMemoryElement();

        }

        public _CMOFExtension CMOFExtension = new _CMOFExtension();

        public class _Extension
        {
            public class _Tag
            {
                public static string @name = "name";
                public IElement _name = null;

                public static string @value = "value";
                public IElement _value = null;

                public static string @element = "element";
                public IElement _element = null;

            }

            public _Tag @Tag = new _Tag();
            public IElement @__Tag = new InMemoryElement();

        }

        public _Extension Extension = new _Extension();

        public class _Common
        {
            public class _ReflectiveSequence
            {
            }

            public _ReflectiveSequence @ReflectiveSequence = new _ReflectiveSequence();
            public IElement @__ReflectiveSequence = new InMemoryElement();

            public class _ReflectiveCollection
            {
            }

            public _ReflectiveCollection @ReflectiveCollection = new _ReflectiveCollection();
            public IElement @__ReflectiveCollection = new InMemoryElement();

        }

        public _Common Common = new _Common();

        public class _CMOF
        {
        }

        public _CMOF CMOF = new _CMOF();

        public class _CMOFReflection
        {
            public class _Factory
            {
            }

            public _Factory @Factory = new _Factory();
            public IElement @__Factory = new InMemoryElement();

            public class _Element
            {
            }

            public _Element @Element = new _Element();
            public IElement @__Element = new InMemoryElement();

            public class _Argument
            {
                public static string @name = "name";
                public IElement _name = null;

                public static string @value = "value";
                public IElement _value = null;

            }

            public _Argument @Argument = new _Argument();
            public IElement @__Argument = new InMemoryElement();

            public class _Extent
            {
            }

            public _Extent @Extent = new _Extent();
            public IElement @__Extent = new InMemoryElement();

            public class _Link
            {
                public static string @firstElement = "firstElement";
                public IElement _firstElement = null;

                public static string @secondElement = "secondElement";
                public IElement _secondElement = null;

                public static string @association = "association";
                public IElement _association = null;

            }

            public _Link @Link = new _Link();
            public IElement @__Link = new InMemoryElement();

            public class _Exception
            {
                public static string @objectInError = "objectInError";
                public IElement _objectInError = null;

                public static string @elementInError = "elementInError";
                public IElement _elementInError = null;

                public static string @description = "description";
                public IElement _description = null;

            }

            public _Exception @Exception = new _Exception();
            public IElement @__Exception = new InMemoryElement();

        }

        public _CMOFReflection CMOFReflection = new _CMOFReflection();

        public class _Reflection
        {
            public class _Factory
            {
                public static string @package = "package";
                public IElement _package = null;

            }

            public _Factory @Factory = new _Factory();
            public IElement @__Factory = new InMemoryElement();

            public class _Type
            {
            }

            public _Type @Type = new _Type();
            public IElement @__Type = new InMemoryElement();

            public class _Object
            {
            }

            public _Object @Object = new _Object();
            public IElement @__Object = new InMemoryElement();

            public class _Element
            {
                public static string @metaclass = "metaclass";
                public IElement _metaclass = null;

            }

            public _Element @Element = new _Element();
            public IElement @__Element = new InMemoryElement();

        }

        public _Reflection Reflection = new _Reflection();

        public static _MOF TheOne = new _MOF();

    }

}

File: src/DatenMeister.SourceGeneration.Console/mof.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports._MOF = void 0;
// Created by DatenMeister.SourcecodeGenerator.TypeScriptInterfaceGenerator Version 1.0.0.0
var _MOF;
(function (_MOF) {
    var _Identifiers;
    (function (_Identifiers) {
        var _URIExtent = /** @class */ (function () {
            function _URIExtent() {
            }
            return _URIExtent;
        }());
        _Identifiers._URIExtent = _URIExtent;
        _Identifiers.__URIExtent_Uri = "dm:///_internal/model/mof#_MOF-Identifiers-URIExtent";
        var _Extent = /** @class */ (function () {
            function _Extent() {
            }
            return _Extent;
        }());
        _Identifiers._Extent = _Extent;
        _Identifiers.__Extent_Uri = "dm:///_internal/model/mof#_MOF-Identifiers-Extent";
    })(_Identifiers = _MOF._Identifiers || (_MOF._Identifiers = {}));
    var _CMOFExtension;
    (function (_CMOFExtension) {
        var _Tag = /** @class */ (function () {
            function _Tag() {
            }
            _Tag.tagOwner = "tagOwner";
            return _Tag;
        }());
        _CMOFExtension._Tag = _Tag;
        _CMOFExtension.__Tag_Uri = "dm:///_internal/model/mof#_MOF-CMOFExtension-Tag";
    })(_CMOFExtension = _MOF._CMOFExtension || (_MOF._CMOFExtension = {}));
    var _Extension;
    (function (_Extension) {
        var _Tag = /** @class */ (function () {
            function _Tag() {
            }
            _Tag._name_ = "name";
            _Tag.value = "value";
            _Tag.element = "element";
            _Tag.metaclass = "metaclass";
            return _Tag;
        }());
        _Extension._Tag = _Tag;
        _Extension.__Tag_Uri = "dm:///_internal/model/mof#_MOF-Extension-Tag";
    })(_Extension = _MOF._Extension || (_MOF._Extension = {}));
    var _Common;
    (function (_Common) {
        var _ReflectiveSequence = /** @class */ (function () {
            function _ReflectiveSequence() {
            }
            return _ReflectiveSequence;
        }());
        _Common._ReflectiveSequence = _ReflectiveSequence;
        _Common.__ReflectiveSequence_Uri = "dm:///_internal/model/mof#_MOF-Common-ReflectiveSequence";
        var _ReflectiveCollection = /** @class */ (function () {
            function _ReflectiveCollection() {
            }
            return _ReflectiveCollection;
        }());
        _Common._ReflectiveCollection = _ReflectiveCollection;
        _Common.__ReflectiveCollection_Uri = "dm:///_internal/model/mof#_MOF-Common-ReflectiveCollection";
    })(_Common = _MOF._Common || (_MOF._Common = {}));
    var _CMOFReflection;
    (function (_CMOFReflection) {
        var _Factory = /** @class */ (function () {
            function _Factory() {
            }
            return _Factory;
        }());
        _CMOFReflection._Factory = _Factory;
        _CMOFReflection.__Factory_Uri = "dm:///_internal/model/mof#_MOF-CMOFReflection-Factory";
        var _Element = /** @class */ (function () {
            function _Element() {
            }
            return _Element;
        }());
        _CMOFReflection._Element = _Element;
        _CMOFReflection.__Element_Uri = "dm:///_internal/model/mof#_MOF-CMOFReflection-Element";
        var _Argument = /** @class */ (function () {
            function _Argument() {
            }
            _Argument._name_ = "name";
            _Argument.value = "value";
            return _Argument;
        }());
        _CMOFReflection._Argument = _Argument;
        _CMOFReflection.__Argument_Uri = "dm:///_internal/model/mof#_MOF-CMOFReflection-Argument";
        var _Extent = /** @class */ (function () {
            function _Extent() {
            }
            return _Extent;
        }());
        _CMOFReflection._Extent = _Extent;
        _CMOFReflection.__Extent_Uri = "dm:///_internal/model/mof#_MOF-CMOFReflection-Extent";
        var _Link = /** @class */ (function () {
            function _Link() {
            }
            _Link.firstElement = "firstElement";
            _Link.secondElement = "secondElement";
            _Link.association = "association";
            return _Link;
        }());
        _CMOFReflection._Link = _Link;
        _CMOFReflection.__Link_Uri = "dm:///_internal/model/mof#_MOF-CMOFReflection-Link";
        var _Exception = /** @class */ (function () {
            function _Exception() {
            }
            _Exception.objectInError = "objectInError";
            _Exception.elementInError = "elementInError";
            _Exception.description = "description";
            return _Exception;
        }());
        _CMOFReflection._Exception = _Exception;
        _CMOFReflection.__Exception_Uri = "dm:///_internal/model/mof#_MOF-CMOFReflection-Exception";
    })(_CMOFReflection = _MOF._CMOFReflection || (_MOF._CMOFReflection = {}));
    var _Reflection;
    (function (_Reflection) {
        var _Factory = /** @class */ (function () {
            function _Factory() {
            }
            _Factory._package_ = "package";
            _Factory.metaclass = "metaclass";
            return _Factory;
        }());
        _Reflection._Factory = _Factory;
        _Reflection.__Factory_Uri = "dm:///_internal/model/mof#_MOF-Reflection-Factory";
        var _Type = /** @class */ (function () {
            function _Type() {
            }
            return _Type;
        }());
        _Reflection._Type = _Type;
        _Reflection.__Type_Uri = "dm:///_internal/model/mof#_MOF-Reflection-Type";
        var _Object = /** @class */ (function () {
            function _Object() {
            }
            return _Object;
        }());
        _Reflection._Object = _Object;
        _Reflection.__Object_Uri = "dm:///_internal/model/mof#_MOF-Reflection-Object";
        var _Element = /** @class */ (function () {
            function _Element() {
            }
            _Element.metaclass = "metaclass";
            return _Element;
        }());
        _Reflection._Element = _Element;
        _Reflection.__Element_Uri = "dm:///_internal/model/mof#_MOF-Reflection-Element";
    })(_Reflection = _MOF._Reflection || (_MOF._Reflection = {}));
})(_MOF = exports._MOF || (exports._MOF = {}));
